Asset/documentdb-dotnet-todo-app/src/Global.asax.cs

```diff
@@ -16,7 +16,7 @@ protected void Application_Start()
             RouteConfig.RegisterRoutes(RouteTable.Routes);
             BundleConfig.RegisterBundles(BundleTable.Bundles);
 
-            DocumentDBRepository<todo.Models.Item>.Initialize();
+            DocumentDBRepository<todo.Models.Item>.InitializeAsync();
```

Comment:
This is no longer blocking and execution will continue before your `DocumentDBRepository` is fully initialized. `System.Web.HttpApplication` really wasn't designed to use async code, while newer ASP.NET models are generally async-only.